includes/course-theme/class-sensei-course-theme-option.php

```diff
@@ -132,31 +144,64 @@ public function should_use_sensei_theme() {
 		return false;
 	}
 
+
+	/**
+	 * Check if it should override the theme for Learning Mode with the course theme.
+	 *
+	 * @return boolean
+	 */
+	public static function should_override_theme() {
+
+		$enabled = ! \Sensei()->settings->get( 'sensei_learning_mode_theme' );
+
+		/**
+		 * Filters if the theme should be overriden for learning mode.
+		 *
+		 * @since 4.0.2
+		 * @hook  sensei_course_learning_mode_theme_override_enabled
+		 *
+		 * @param {bool} $enabled True if the learning mode theme override is enabled.
+		 */
+		return (bool) apply_filters( 'sensei_course_learning_mode_theme_override_enabled', $enabled );
+	}
+
 	/**
-	 * Check if the given course has the Sensei Theme enabled or not.
+	 * Check if the given course has Learning Mode enabled or not.
 	 *
 	 * @param int $course_id Course ID.
 	 *
+	 * @deprecated 4.0.2 Use Sensei_Course_Theme_Option::has_learning_mode_enabled
+	 *
 	 * @return bool
 	 */
 	public static function has_sensei_theme_enabled( $course_id ) {
+		return self::has_learning_mode_enabled( $course_id );
+	}
+
+	/**
+	 * Check if the given course has Learning Mode enabled or not.
+	 *
+	 * @param int $course_id Course ID.
+	 *
+	 * @return bool
+	 */
+	public static function has_learning_mode_enabled( $course_id ) {
 		$theme              = get_post_meta( $course_id, self::THEME_POST_META_NAME, true );
 		$enabled_for_course = self::SENSEI_THEME === $theme;
 		$enabled_globally   = (bool) \Sensei()->settings->get( 'sensei_learning_mode_all' );
 
+		$enabled = $enabled_for_course || $enabled_globally;
+
 		/**
 		 * Filters if a course has learning mode enabled.
 		 *
 		 * @since 4.0.0
-		 * @hook sensei_course_learning_mode_enabled
+		 * @hook  sensei_course_learning_mode_enabled
 		 *
-		 * @param {bool} $enabled_for_course True if the learning mode is enabled for the course.
-		 * @param {bool} $enabled_globally   True if the learning mode is enabled globally.
-		 * @param {int}  $course_id          The id of the course.
+		 * @param {bool} $enabled    True if the learning mode is enabled for the course or globally.
+		 * @param {int}  $course_id  The id of the course.
 		 */
-		$enabled_via_filter = (bool) apply_filters( 'sensei_course_learning_mode_enabled', $enabled_for_course, $enabled_globally, $course_id );
-
-		return $enabled_for_course || $enabled_globally || $enabled_via_filter;
+		return (bool) apply_filters( 'sensei_course_learning_mode_enabled', $enabled, $course_id );
```

Comment:
Since it'd be so complicated and confusing, and it's only been out for a little while, I'd say let's change the existing one. I added it to the updated hook list to be included in the release post, and if anyone is using it already they can easily update it or open an issue.